core/src/services/mysql/backend.rs

```diff
@@ -0,0 +1,317 @@
+                    Error::new(ErrorKind::ConfigInvalid, "connection_string is empty")
+                        .with_context("service", Scheme::Mysql),
+                )
+            }
+        };
+
+        let config = Opts::from_url(&conn).map_err(|err| {
+            Error::new(ErrorKind::ConfigInvalid, "connection_string is invalid")
+                .with_context("service", Scheme::Mysql)
+                .set_source(err)
+        })?;
+
+        let table = match self.table.clone() {
+            Some(v) => v,
+            None => {
+                return Err(Error::new(ErrorKind::ConfigInvalid, "table is empty")
+                    .with_context("service", Scheme::Mysql))
+            }
+        };
+        let key_field = match self.key_field.clone() {
+            Some(v) => v,
+            None => "key".to_string(),
+        };
+        let value_field = match self.value_field.clone() {
+            Some(v) => v,
+            None => "value".to_string(),
+        };
+        let root = normalize_root(
+            self.root
+                .clone()
+                .unwrap_or_else(|| "/".to_string())
+                .as_str(),
+        );
+        let pool = Pool::new(config.clone());
+
+        Ok(MySqlBackend::new(Adapter {
+            connection_pool: pool,
+            config,
+            table,
+            key_field,
+            value_field,
+        })
+        .with_root(&root))
+    }
+}
+
+/// Backend for mysql service
+pub type MySqlBackend = kv::Backend<Adapter>;
+
+#[derive(Clone)]
+pub struct Adapter {
+    connection_pool: Pool,
+    config: Opts,
+
+    table: String,
+    key_field: String,
+    value_field: String,
+}
+
+impl Debug for Adapter {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("Adapter")
+            .field("connection_pool", &self.connection_pool)
+            .field("config", &self.config)
+            .field("table", &self.table)
+            .field("key_field", &self.key_field)
+            .field("value_field", &self.value_field)
+            .finish()
+    }
+}
+
+#[async_trait]
+impl kv::Adapter for Adapter {
+    fn metadata(&self) -> kv::Metadata {
+        kv::Metadata::new(
+            Scheme::Mysql,
+            &self.table,
+            Capability {
+                read: true,
+                write: true,
+                ..Default::default()
+            },
+        )
+    }
+
+    async fn get(&self, path: &str) -> Result<Option<Vec<u8>>> {
+        let query = format!(
+            "SELECT `{}` FROM `{}` WHERE `{}` = :path LIMIT 1",
+            self.value_field, self.table, self.key_field
+        );
+        let mut conn = self.connection_pool.get_conn().await.map_err(|err| {
+            Error::new(ErrorKind::Unexpected, "connection failed")
+                .with_context("service", Scheme::Mysql)
+                .set_source(err)
+        })?;
+        let statement = conn.prep(query).await.map_err(|err| {
```

Comment:
Can we prepare the sql in the backend to speed up the query? (We can improve this part in later PRs.)

Reply:
Seems not; the `async_mysql` library has some design issues. When we use this library, we need to manage the connection from the connection pool manually, and the prepared statement in `async_mysql` is binding with the connection. I think there would be some problem if we cache the statement

Reply:
Got it, thanks!